tests/pos/i8577/Main_2.scala

```diff
@@ -0,0 +1,165 @@
+package i8577
+
+def main: Unit = {
+  {
+    extension (ctx: StringContext) def macroA: MacroB.StringContext = MacroB(ctx)
+    extension (inline ctx: MacroB.StringContext) inline def unapplySeq(inline input: Int): Option[Seq[Int]] =
+      ${implUnapplyA('ctx, 'input)}
+
+    val macroA"$xA" = 1
```

Comment:
will do

Reply:
Although, is it really necessary if I later do `assert(xA == 1)`?

Reply:
The assert and ascription tests are complementary. That ascription will check that at compiletime the binding has the type we expect. On the other hand the assertion will check at runtime that the bound value is correct. 

Note that the patten could bind to a type `Any` and have the value 1. In these simple cases the code will compile an pasa the assertion, but other code that relies on the type of the binding to be more precise it might not compile